OMCompiler/Compiler/BackEnd/ResolveLoops.mo

```diff
@@ -1594,17 +1594,15 @@ protected function priorizeEqsWithVarCrosses2
   input array<list<Integer>> priorities;
 protected
   Boolean b0,b1,b2;
-  Integer numCrossVars;
   list<Integer> eqVars,crossVars;
 algorithm
   eqVars := arrayGet(mIn,eq);
   crossVars := List.intersectionOnTrue(eqVars,varCrossLst,intEq);
-  numCrossVars := listLength(crossVars);
-  if numCrossVars == 0 then
+  if listEmpty(crossVars) then
     arrayGetAppendLst(1,{eq},priorities);
-  elseif numCrossVars == 1 then
+  elseif listLength(crossVars) == 1 then
     arrayGetAppendLst(2,{eq},priorities);
-  elseif numCrossVars >= 2 then
+  else
     arrayGetAppendLst(3,{eq},priorities);
```

Comment:
Doing `listLength` on an empty list is pretty much the same as doing `listEmpty`, so there's not much point in first checking if the list is empty here. But it probably doesn't really matter one way or another.

Reply:
Yes, I didn't mention that but we already have `List.hasOneElement` that you could use here.

Reply:
Up to you, it doesn't really matter.

Reply:
Ok. I replaced this `listLength() == 1` by `List.hasOneElement()`. I will open a second PR with the other changes.